File: android/widget/webview/src/org/opentravelmate/widget/webview/js/WebViewJavaObject.java
package org.opentravelmate.widget.webview.js;

import android.webkit.JavascriptInterface;

/**
 * Implementation of the webViewJavaObject interface.
 * 
 * @author Marc Plouhinec
 */
public class WebViewJavaObject {
	
	/**
	 * Get the root web view ID.
	 * 
	 * @return web view ID.
	 */
	@JavascriptInterface
	public String getRootWebViewId() {
		return "FAKE-ID";
	}
	
	public void updatePlaceHolderLayoutParams(int windowWidth, int windowHeight, Object placeholderLayoutParams) {
		System.out.println("windowWidth = " + windowWidth);
		System.out.println("windowHeight = " + windowHeight);
		System.out.println("placeholderLayoutParams = " + placeholderLayoutParams);
	}

	public void registerWebViewClass(Object webViewClass) {
		System.out.println("webViewClass = " + webViewClass);
	}
}

File: android/widget/webview/resources/js/org/opentravelmate/widget/webview/v1/webViewJavaObject.js
/**
 * Interface implemented by a Java Object
 *
 * @author marc.plouhinec@opentravelmate.org (Marc Plouhinec)
 */


define([], function() {

    var webViewJavaObject = {};

  /**
   * Get the root web view ID.
   *
   * @return {string} web view ID.
   */
  webViewJavaObject.getRootWebViewId = function() {
    return '';
  };

  /**
   * Update the place-holders layout parameters.
   *
   * @param {number} windowWidth Window width.
   * @param {number} windowHeight Window height.
   * @param {object} placeholderLayoutParams Place-holders layout parameters.
   */
  webViewJavaObject.updatePlaceHolderLayoutParams = function(windowWidth, windowHeight, placeholderLayoutParams) {
  };

  return webViewJavaObject;
});
